investigations/Possible_Instrument_View_Improvements.md

```diff
@@ -0,0 +1,48 @@
+
+### Caching Layer Lookup vs Detector Object Access
+Following on from the last point, in most places where information about individual detectors are required, a pointer/reference to the detector object is obtained in order to retrieve detector id, shape, bounding boxes and corresponding workspace data. Instrument 2.0 simplifies this access by making most of this information directly accessible from the `DetectorInfo`/`ComponentInfo`. 
+
+### List of Operations Instrument View requires of Instrument 1.0
+- Parametrised instrument constructor 
+- `Instrument::getDetector()`
+- `Instrument::getSample()`
+- `Instrument::getSource()`
+- `Instrument::getDetectorIDs()`
+- `Instrument::getBoundingBox()`
+- `Instrument::getPhysicalInstrument()`
+- `Instrument::setPhysicalInstrument()`
+- `Detector::getRotation()`
+- `Detector::getPos()`
+- `Detector::getID()`
+- `Component::getParent()`
+- `Component::getComponentID()`
+- `ObjComponent::getBoundingBox()`
+- `ObjComponent::shape()`
+- `ObjComponent::draw()`
+
+### Considerations
+- Is the current geometry renderer the best solution for Instrument View? Should the rendering be decoupled from the objects and abstracted away at a higher level? Perhaps a renderer which takes an instrument (`DetectorInfo`/`ComponentInfo`)and renders all (or a subset of) the components. Seems like visualization is mixed with in-memory representation. - ***"A renderer class which accepts a ComponentInfo and a list of component indices could be a good solution in order to create a more sensible code layout."***
+- Why is `PickID` stored as some strange encoded colour? Couldn't some kind of map be built using the `DetectorID` directly so that on mouse over of a `ProjectionSurface` one could immediately retrieve the `DetectorID`? - ***"The pick color is a convenient way of addressing detectors using only 3 bytes (RGB ingoring Alpha) in OpenGL space. It makes it easier to handle mouse over on the visualized geometry. Detector IDs are sufficiently large to exceed the three bytes."***
+- Will workspace indices be directly accessible from `DetectorInfo`? ***"Not necessarily, however, a map between detector indices/IDs can be built and used if necessary. Also, there is SpectrumInfo which maps spectrum index to detector indices in the SpectrumDefinition.***
+
+### Recommendations
+
+#### Calls to ExperimentInfo::getInstrument()
+
+It seems as though calls to `ExperimentInfo::getInstrument()` are remnants of a legacy system where instrument geometry and associated parameter maps were always calculated on the fly. Since geometry is stored (in both Instrument 1.0 and 2.0) and the workspace observer mechanism allows for response to workspace updates, this is not at all necessary. These should be removed in favour of storing the `ComponentInfo/DetectoInfo`
+
```

Comment:
const reference (alias) to `ComponentInfo/DetectorInfo`. Very important that this statement is not interpreted as a copy of either of these types.